src/api.ts

```diff
@@ -208,7 +208,8 @@ class RippleAPI extends EventEmitter {
       lastBatchLength = singleResult[collectKey].length
       results.push(singleResult)
     } while(!!marker && count < countTo && lastBatchLength !== 0)
-    return results
+    // NOTE: We slice out any additional items over the original requested num
+    return results.slice(0, countTo)
```

Comment:
follow up: this was reverted for now. Since `results` is an array of full response objects (where responses can contain multiple data objects).